samcli/local/docker/lambda_image.py

```diff
@@ -328,6 +328,6 @@ def is_rapid_image(image_name: str) -> bool:
         : return bool: True, if the image name ends with rapid-$SAM_CLI_VERSION. False, otherwise
         """
 
-        if not re.search(fr":{RAPID_IMAGE_TAG_PREFIX}-\d+\.\d+.\d+$", image_name):
+        if not re.search(fr":{RAPID_IMAGE_TAG_PREFIX}-\d+\.\d+\.[\w\.]+$", image_name):
```

Comment:
Could we just check that it starts with `rapid-`?

Reply:
Proposal:
```
        try:
            return image_name.split(":")[1].startswith(f"{RAPID_IMAGE_TAG_PREFIX}-")
        except (IndexError, AttributeError):
            # split() returned 1 or less items or image_name is None
            return False
```

Also adding tests in `tests/unit/local/docker/test_lambda_image.py`:
```
    def test_is_rapid_image(self):
        self.assertFalse(LambdaImage.is_rapid_image(None))
        self.assertFalse(LambdaImage.is_rapid_image(""))
        self.assertFalse(LambdaImage.is_rapid_image("my_repo"))
        self.assertFalse(LambdaImage.is_rapid_image("my_repo:tag"))

        self.assertTrue(LambdaImage.is_rapid_image("my_repo:rapid-1.29beta"))
        self.assertTrue(
            LambdaImage.is_rapid_image(f"public.ecr.aws/sam/emulation-python3.6:rapid-1.29.0")
        )
```